diff --git a/tests/test_requests.py b/tests/test_requests.py
index d8fbb236..20314bfe 100644
--- a/tests/test_requests.py
+++ b/tests/test_requests.py
@@ -2697,7 +2697,7 @@ class TestPreparingURLs:
             b"http://*",
             "http://*.google.com",
             "http://*",
-            "http://☃.net/",
+            # "http://☃.net/",  qh3 idna encoder accept, firefox accept, native idna + strict refuse.
         ),
     )
     def test_preparing_bad_url(self, url):
